packages/xrpl/src/Wallet/index.ts

```diff
@@ -337,11 +345,17 @@ class Wallet {
       )
     }
 
-    removeTrailingZeros(tx)
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Only modifies Payments
+    removeTrailingZeros(tx as Transaction)
 
     const txToSignAndEncode = { ...tx }
 
     txToSignAndEncode.SigningPubKey = multisignAddress ? '' : this.publicKey
+    if (tx.SigningPubKey && txToSignAndEncode.SigningPubKey === '') {
```

Comment:
Was this validation needed to added custom defintions?

Reply:
No, I'm honestly not sure why I added this. Maybe it was meant to be part of another PR but I was on the wrong branch? I'll remove it for now.